src/test/java/uk/nickbdyer/httpserver/controllers/FileControllerTest.java

```diff
@@ -50,4 +55,32 @@ public void willReturnFileContentsInBody() throws IOException {
         assertEquals("hello", new String(response.getBody()));
     }
 
+    @Test
+    public void willReturn204ForValidPatchEtag() throws IOException {
```

Comment:
This test reinforces my argument from above re. "Too much low level in controller". 

To make sure that the sha1 matching works, I should not need requests, responses etc. this feels complicated.